git/git.go

```diff
@@ -714,6 +714,63 @@ func CloneWithoutFilters(args []string) error {
 	return nil
 }
 
+// RemoteRefs returns a list of branches & tags for a remote
+// If retrieveFromRemote is true, this information is built by actually calling
+// the remote using 'git ls-remote'. Otherwise the cached remote branch
+// information held locally is used instead.
+func RemoteRefs(remoteName string, retrieveFromRemote bool) ([]*Ref, error) {
+
```

Comment:
> I think I'd rather have two separate functions, RemoteRefs() and CachedRemoteRefs()

Yeah, fair enough.

> I'd also prefer the scanner loop use continue, instead of continuing to nest if statements

Hrm, double-edged sword IMO, continue is a pseudo-GOTO in many ways and can have it's own problems if over-used. But it's a short loop body so ok.